File: fintechplatform/src/main/java/com/fintechplatform/android/profile/jobinfo/ui/JobInfoViewComponent.java
package com.fintechplatform.android.profile.jobinfo.ui;

import com.fintechplatform.android.alert.AlertHelpersModule;
import com.fintechplatform.android.api.NetModule;
import com.fintechplatform.android.log.LogModule;
import com.fintechplatform.android.profile.api.ProfileAPIModule;
import com.fintechplatform.android.profile.db.user.UsersPersistanceDBModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component (modules = {
        JobInfoPresenterModule.class,
        NetModule.class,
        ProfileAPIModule.class,
        LogModule.class,
        AlertHelpersModule.class,
        UsersPersistanceDBModule.class,

})
public interface JobInfoViewComponent {
    void inject(JobInfoActivity activity);
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/jobinfo/di/JobInfoViewComponent.kt
package com.fintechplatform.ui.profile.jobinfo.di

import com.fintechplatform.api.log.LogModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.api.profile.api.IdsDocumentsAPIModule
import com.fintechplatform.api.profile.api.ProfileAPIModule
import com.fintechplatform.ui.alert.AlertHelpersModule
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDBModule
import com.fintechplatform.ui.profile.jobinfo.JobInfoFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [
    JobInfoPresenterModule::class, 
    NetModule::class, 
    ProfileAPIModule::class, 
    IdsDocumentsAPIModule::class,
    LogModule::class, 
    AlertHelpersModule::class, 
    UsersPersistanceDBModule::class
])
interface JobInfoViewComponent {
    fun inject(fragment: JobInfoFragment)
}
